stream-chat-android/src/main/java/com/getstream/sdk/chat/viewmodel/channels/ChannelsViewModel.kt

```diff
@@ -22,13 +22,13 @@ public class ChannelsViewModel(
     private val limit: Int = 30
 ) : ViewModel() {
     private val channelsData: LiveData<State>
-    private val loadingMoreData: LiveData<State.LoadingNextPage>
     private val loadingData = MutableLiveData<State.Loading>()
-    private val endPageData: LiveData<State.EndPageReached>
     private val stateMerger = MediatorLiveData<State>()
-
     public val state: LiveData<State> = stateMerger
 
+    private val paginationStateMerger = MediatorLiveData<PaginationState>()
+    public val paginationState: LiveData<PaginationState> = paginationStateMerger
```

Comment:
I'd like to use it inside the main state of the screen as part of Result state with list of channels. It will let us to keep an MVI architecture

Reply:
Then, I guess, we will need to rewrite our neat sealed classes to something more generic like a single state data class

Reply:
We need to test DiffUtil performance and decide if we can afford to execute it frequently. For example, when we reach the end of the channel list we observe new state:

1. When loading more is set to true
2. When new chunk of channels arrive
3. When loading more is set to false
4. When `endOfChannels` emits true

Theoretically this can be reduced to just 2 emits: 
1. When loading more starts (probably displaying progress indicator)
2. When data arrives together with with `endOfChannels`flag

Reply:
Your version works fine, it's just my sense of architecture approach and coding ideas. Let's discuss with others